From 5271f61efe976ab08df4f78e04f0c0a8d524d5fb Mon Sep 17 00:00:00 2001
From: Rene Kita <mail@rkta.de>
Date: Sat, 26 Nov 2022 10:35:09 +0100
Subject: [PATCH 170/172] Remove const where we can't keep the promise

In all that cases compiling with -Wcast-qual shows, that the const was
cast away and there was no easy way to keep the const.
---
 Str.c   |    2 +-
 file.c  |    4 ++--
 fm.h    |    2 +-
 form.c  |    2 +-
 func.c  |    6 +++---
 proto.h |   11 +++++------
 rc.c    |    8 ++++----
 url.c   |   19 +++++++++----------
 8 files changed, 26 insertions(+), 28 deletions(-)

--- a/file.c
+++ b/file.c
@@ -1752,7 +1752,7 @@ loadGeneralFile(char *path, ParsedURL *v
 
   load_doc:
     {
-	const char *sc_redirect;
+	char *sc_redirect;
 	parseURL2(tpath, &pu, current);
 	sc_redirect = query_SCONF_SUBSTITUTE_URL(&pu);
 	if (sc_redirect && *sc_redirect && checkRedirection(&pu)) {
@@ -7968,7 +7968,7 @@ loadImageBuffer(URLFile *uf, Buffer *new
     URLFile f;
     MySignalHandler(*volatile prevtrap) (SIGNAL_ARG) = NULL;
     struct stat st;
-    const ParsedURL *pu = newBuf ? &newBuf->currentURL : NULL;
+    ParsedURL *pu = newBuf ? &newBuf->currentURL : NULL;
 
     loadImage(newBuf, IMG_FLAG_STOP);
     image.url = uf->url;
--- a/fm.h
+++ b/fm.h
@@ -278,7 +278,7 @@ extern int REV_LB[];
 #define SCONF_NO_REFERER_TO	4
 #define SCONF_USER_AGENT	5
 #define SCONF_N_FIELD		6
-#define query_SCONF_SUBSTITUTE_URL(pu) ((const char *)querySiteconf(pu, SCONF_SUBSTITUTE_URL))
+#define query_SCONF_SUBSTITUTE_URL(pu) ((char *)querySiteconf(pu, SCONF_SUBSTITUTE_URL))
 #define query_SCONF_USER_AGENT(pu) ((const char *)querySiteconf(pu, SCONF_USER_AGENT))
 #define query_SCONF_URL_CHARSET(pu) ((const wc_ces *)querySiteconf(pu, SCONF_URL_CHARSET))
 #define query_SCONF_NO_REFERER_FROM(pu) ((const int *)querySiteconf(pu, SCONF_NO_REFERER_FROM))
--- a/form.c
+++ b/form.c
@@ -904,7 +904,7 @@ loadPreForm(void)
 	s = getWord(&p);
 
 	if (!strcmp(s, "url")) {
-	    arg = getRegexWord((const char **)&p, &re_arg);
+	    arg = getRegexWord(&p, &re_arg);
 	    if (!arg || !*arg)
 		continue;
 	    p = getQWord(&p);
--- a/func.c
+++ b/func.c
@@ -446,10 +446,10 @@ getQWord(char **str)
  * XXX: Actually this is unrelated to func.c.
  */
 char *
-getRegexWord(const char **str, Regex **regex_ret)
+getRegexWord(char **str, Regex **regex_ret)
 {
     char *word = NULL;
-    const char *p, *headp, *bodyp, *tailp;
+    char *p, *headp, *bodyp, *tailp;
     char delimiter;
     int esc;
     int igncase = 0;
@@ -514,7 +514,7 @@ getRegexWord(const char **str, Regex **r
 
 not_regex:
     p = headp;
-    word = getQWord((char **)&p);
+    word = getQWord(&p);
     if (regex_ret)
 	*regex_ret = NULL;
 
--- a/proto.h
+++ b/proto.h
@@ -172,11 +172,10 @@ extern void chkExternalURIBuffer(Buffer
 #endif
 extern ParsedURL *schemeToProxy(int scheme);
 #ifdef USE_M17N
-extern wc_ces url_to_charset(const char *url, const ParsedURL *base,
+extern wc_ces url_to_charset(char *url, ParsedURL *base,
 			     wc_ces doc_charset);
-extern char *url_encode(const char *url, const ParsedURL *base,
-			wc_ces doc_charset);
-extern char *url_decode2(const char *url, const Buffer *buf);
+extern char *url_encode(char *url, ParsedURL *base, wc_ces doc_charset);
+extern char *url_decode2(char *url, Buffer *buf);
 #else /* !defined(USE_M17N) */
 #define url_encode(url, base, cs) url_quote(url)
 extern char *url_decode0(const char *url);
@@ -615,7 +614,7 @@ extern char *etcFile(char *base);
 extern char *confFile(char *base);
 extern char *auxbinFile(char *base);
 extern char *helpFile(char *base);
-extern const void *querySiteconf(const ParsedURL *query_pu, int field);
+extern void *querySiteconf(ParsedURL *query_pu, int field);
 extern Str localCookie(void);
 extern Str loadLocalDir(char *dirname);
 extern void set_environ(char *var, char *value);
@@ -731,7 +730,7 @@ extern char *getKeyData(int key);
 extern char *getWord(char **str);
 extern char *getQWord(char **str);
 struct regex;
-extern char *getRegexWord(const char **str, struct regex **regex_ret);
+extern char *getRegexWord(char **str, struct regex **regex_ret);
 #ifdef USE_MOUSE
 extern void initMouseAction(void);
 #endif
--- a/rc.c
+++ b/rc.c
@@ -1798,7 +1798,7 @@ loadSiteconf(void)
 
 	    /* Second, create a new record. */
 	    newent = newSiteconfRec();
-	    url = getRegexWord((const char **)&p, &newent->re_url);
+	    url = getRegexWord(&p, &newent->re_url);
 	    opt = getWord(&p);
 	    SKIP_BLANKS(p);
 	    if (!newent->re_url) {
@@ -1859,10 +1859,10 @@ loadSiteconf(void)
     fclose(fp);
 }
 
-const void *
-querySiteconf(const ParsedURL *query_pu, int field)
+void *
+querySiteconf(ParsedURL *query_pu, int field)
 {
-    const struct siteconf_rec *ent;
+    struct siteconf_rec *ent;
     Str u;
     char *firstp, *lastp;
 
--- a/url.c
+++ b/url.c
@@ -2460,14 +2460,14 @@ schemeToProxy(int scheme)
 
 #ifdef USE_M17N
 wc_ces
-url_to_charset(const char *url, const ParsedURL *base, wc_ces doc_charset)
+url_to_charset(char *url, ParsedURL *base, wc_ces doc_charset)
 {
-    const ParsedURL *pu;
+    ParsedURL *pu;
     ParsedURL pu_buf;
     const wc_ces *csptr;
 
     if (url && *url && *url != '#') {
-	parseURL2((char *)url, &pu_buf, (ParsedURL *)base);
+	parseURL2(url, &pu_buf, base);
 	pu = &pu_buf;
     } else {
 	pu = base;
@@ -2480,23 +2480,22 @@ url_to_charset(const char *url, const Pa
 }
 
 char *
-url_encode(const char *url, const ParsedURL *base, wc_ces doc_charset)
+url_encode(char *url, ParsedURL *base, wc_ces doc_charset)
 {
-    return url_quote_conv((char *)url,
-			  url_to_charset(url, base, doc_charset));
+    return url_quote_conv(url, url_to_charset(url, base, doc_charset));
 }
 
 char *
-url_decode2(const char *url, const Buffer *buf)
+url_decode2(char *url, Buffer *buf)
 {
     wc_ces url_charset;
 
     if (!DecodeURL)
-	return (char *)url;
+	return url;
     url_charset = buf ?
-	url_to_charset(url, baseURL((Buffer *)buf), buf->document_charset) :
+	url_to_charset(url, baseURL(buf), buf->document_charset) :
 	url_to_charset(url, NULL, 0);
-    return url_unquote_conv((char *)url, url_charset);
+    return url_unquote_conv(url, url_charset);
 }
 
 #else /* !defined(USE_M17N) */
--- a/Str.c
+++ b/Str.c
@@ -169,7 +169,7 @@ Strcopy(Str x, Str y)
 	    exit(1);
 	x->area_size = y->length + 1;
     }
-    bcopy((void *)y->ptr, (void *)x->ptr, y->length + 1);
+    bcopy((const void *)y->ptr, (void *)x->ptr, y->length + 1);
     x->length = y->length;
 }
 
